pyspecProcScripts/integrate_limits.py

```diff
@@ -1,43 +1,148 @@
+
+    Gaussian: boolean
+        set `Lorentzian` to True in order to calculate the matched-filter via
+        Lorentzian convolution of the frequency domain signal (i.e., apply a
+        matched Lorentzian filter to peak in frequency domain).
+        If `Lorentzian` is set to True, set `Gaussian` to False.
+
+
+    fl: None or figlist_var()
+    `   to show diagnostic plots, set `fl` to the figure list; set `fl` to None in
+        order not to see any diagnostic plots
 
-    Returns
-    =======
-    retval: ndarray
-        A len 2 ndarray tuple with the start and stop indeces for the integration
-        bounds.
     """
-    signal_sign = s.C.sum(axis).run(np.real).run(np.sign)
-    temp = abs(s).real * signal_sign
-
-    # pulled from apodization code
-    sigma = nddata(np.linspace(1e-5,1e3,1000),'sigma').set_units('sigma','s')
-    s_avg = s.C.mean_all_but('t2')
-    gaussians = np.exp(-s_avg.C.fromaxis('t2')**2/2/sigma**2)
-    signal_E = (abs(s_avg * gaussians)**2).sum('t2')
-    signal_E /= signal_E.data.max()
-    filter_width = abs(signal_E-1/sqrt(2)).argmin('sigma').item()
+    temp = s.C.mean_all_but(axis)
     if fl is not None:
+        fl.next('integration diagnostic')
         fl.push_marker()
-        fl.next('signal Energy')
+        fl.plot(abs(temp.C.ft('t2'))/abs(temp.C.ft('t2')).max(), alpha=0.6, label='before convolve')
+    Gaussian_Conv = Gaussian
+    Lorentzian_Conv = Lorentzian
+    Lorentz_to_Gauss = False
+    #{{{ make sure to calculate Lorentzian filter for L-to-G transform
+    if Lorentz_to_Gauss:
+        assert Lorentzian_Conv == True and Gaussian_Conv == False, "Must set Lorentzian convolve to True for Lorentzian to Gaussian transformation"
+    #}}}
+    # Calculating matched filter
+    sigma = nddata(np.linspace(1e-10,1e-1,1000),'sigma').set_units('sigma','s')
+    if Gaussian_Conv:
+        gaussians = np.exp(-temp.C.fromaxis(axis)**2/2/sigma**2)
+        signal_E = (abs(temp * gaussians)**2).sum(axis)
+    if Lorentzian_Conv:
+        lorentzians = np.exp(-abs(temp.C.fromaxis(axis))/sigma)
+        signal_E = (abs(temp * lorentzians)**2).sum(axis)
+    signal_E /= signal_E.data.max()
+    if Gaussian_Conv:
+        filter_width = abs(signal_E-1/sqrt(2)).argmin('sigma').item()
+    if Lorentzian_Conv:
+        filter_width = abs(signal_E-signal_E.max()/2).argmin('sigma').item()
+    if fl is not None:
+        fl.next('integration diagnostic -- signal Energy')
         fl.plot(signal_E, human_units=False)
         fl.plot(signal_E['sigma':(filter_width,filter_width+1e-6)],'o', human_units=False)
+    print("*** *** ***")
+    print("FILTER WIDTH IS",filter_width)
+    print("*** *** ***")
+    # Generate the appropriate convolution functions
+    Gaussian_func = lambda x,y: exp(-(x**2)/(2.0*(y**2)))
+    Lorentzian_func = lambda x,y: exp(-(abs(x))/y)
+    if Gaussian_Conv:
+        convfunc = Gaussian_func
+    if Lorentzian_Conv:
+        convfunc = Lorentzian_func
```

Comment:
defs should just go inside the conditionals -- second one should be elif

Reply:
they are defined outside the conditionals for the Lorentz to Gauss part

Reply:
yes I'm just saying that they should go inside -- unless I'm missing something and they are used elsewhere

(Why define the Gaussian (Lorentzian) function if you're not going to use it.)

Reply:
The only reason it is like that is the Lorentzian-to-Gaussian. This boolean controls the method used to calculate the filter width, based on the energies. For Loretzian-to-Gaussian, I want to use the filter width for the Lorentzian (so I set Lorentzian to True), but I also convolve with a Gaussian (before dividing by the Lorentzian), and so rather than define the Gaussian twice in the same script, I set it outside the booleans so it can get called when it is needed, either for 1. Gaussian convolution or 2. Lorentzian-to-Gaussian